vktrace/vktrace_layer/vktrace_lib_trace.cpp

```diff
@@ -953,6 +953,13 @@ VKTRACER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL __HOOKED_vkCreateDevice(VkPhysica
     // remove the loader extended createInfo structure
     VkDeviceCreateInfo localCreateInfo;
     memcpy(&localCreateInfo, pCreateInfo, sizeof(localCreateInfo));
+
```

Comment:
I think that these new lines aren't needed.  

A better fix might be to simply remove lines 956-963.  `add_VkDeviceCreateInfo_to_packet()` is probably the only thing that references these lists of strings and all it does is copy the strings into the trace buffer.  The existing arrays of char* owned by the app can still be used for this since they don't need to be modified.

The existing code at 956-963 is modifying the app's memory when it doesn't need to.  And it is just copying the pointers back on top of themselves anyway.

The only reason to keep the existing code (and add the suggested lines) is if the lists of extensions or layers can be  modified someplace that I may have missed.